### PR TITLE
Batched Quantized Matmul + Fast Small QMV

### PR DESCRIPTION
**Pull request by barronalex**
- Support batched inputs for quantized matmuls
- Add a QMV specialization for `D=64,128`

The batched `qmv` and `qvm` will be useful for a quantized KV cache.

The existing `qmv` was pretty slow for `D=128` which is by far the most common attention head dim in LLMs.
This is because if you're using `simd_sum` for `D=128` you're only processing 4 values for each of the 32 threads in your simdgroup. This isn't enough to max out the memory bandwidth and since that's the main bound for `qmv`, you sacrifice a lot of performance.

Instead, the new specialization uses `quad_sum` over quadgroups of 4 threads. This means each thread processes 32 values and the overall throughput is much higher.

On M1 Max:
```
queries = (1, 8, 4, 1, 128)
keys = (1, 8, 4, 32768, 128)
```
Before:
```
Timing matmul ... 1.68047 msec
Timing quant_matmul ... 2.66873 msec
```
After:
```
Timing matmul ... 1.67618 msec
Timing quant_matmul ... 0.54323 msec
```

### CODE REVIEW

**Comment by barronalex** [comment]
Looks like we were slightly regressing performance so I added a `batched` template param.
Now seems like we're at parity with `main`:

On M2 Ultra with a 3B model
Before: `180.6` tks (avg of 3)
After:    `180.6` tks (avg of 3)

**Review by angeloskath — APPROVED** [review]
This is absolutely fantastic!

I left one tiny comment on the op. Everything else looks great!

File: mlx/ops.cpp [hunk 1]
```diff
@@ -3592,23 +3592,44 @@ array conv_general(
...
+    const array& in_scales,
+    const array& in_biases,
     bool transpose /* = true */,
     int group_size /* = 64 */,
     int bits /* = 4 */,
     StreamOrDevice s /* = {} */) {
+  auto x = in_x;
+  auto w = in_w;
+  auto scales = in_scales;
+  auto biases = in_biases;
```
[the remark this answers is not in the document]
Comment (barronalex) [reply mlx/ops.cpp:3606]: Nice! Will do

File: mlx/backend/metal/quantized.cpp
[the remark this answers is not in the document]
Comment (barronalex) [reply mlx/backend/metal/quantized.cpp:1]: Agreed. I'm going to just bite the bullet and do it in this PR. It's been a bit painful but it's almost done.

**Pull request merged**